### ISSUE TITLE
Carthage Update Failed

### ISSUE THREAD

**Issue by ejimenez93** [CosmicMind/Material#600]
Getting this error after running `carthage update --platform ios`. Using XCode 8 and Carthage v0.18.1

```*** Fetching Material
*** Checking out Material at "2.3.10"
*** xcodebuild output can be found in /var/folders/r9/s0kfy3cd76z77gm4qt6y1qx00000gn/T/carthage-xcodebuild.xPo7cN.log
*** Building scheme "Material iOS" in Material.xcodeproj
** BUILD FAILED **


The following build commands failed:
	CompileSwift normal arm64
	CompileSwiftSources normal arm64 com.apple.xcode.tools.swift.compiler
(2 failures)
/Carthage/Checkouts/Material/Sources/iOS/NavigationDrawerController.swift:1087:15: error: Objective-C method 'gestureRecognizer:shouldReceive:' provided by method 'gestureRecognizer(_:shouldReceive:)' does not match the requirement's selector ('gestureRecognizer:shouldReceiveTouch:')
A shell task (/usr/bin/xcrun xcodebuild -project /Carthage/Checkouts/Material/Material.xcodeproj -scheme "Material iOS" -configuration Release -sdk iphoneos ONLY_ACTIVE_ARCH=NO BITCODE_GENERATION_MODE=bitcode CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES clean build) failed with exit code 65:
** BUILD FAILED **


The following build commands failed:
	CompileSwift normal arm64
	CompileSwiftSources normal arm64 com.apple.xcode.tools.swift.compiler
(2 failures)```

**Comment by ejimenez93** [comment]
Hey @danieldahan,

Thanks for the response! Unfortunately that didn't work:

```
*** Fetching Material
*** Checking out Material at "2.3.11"
*** xcodebuild output can be found in /var/folders/r9/s0kfy3cd76z77gm4qt6y1qx00000gn/T/carthage-xcodebuild.fKIVtj.log
*** Building scheme "Material iOS" in Material.xcodeproj
** BUILD FAILED **


The following build commands failed:
	CompileSwift normal arm64
	CompileSwiftSources normal arm64 com.apple.xcode.tools.swift.compiler
(2 failures)
/Carthage/Checkouts/Material/Sources/iOS/NavigationDrawerController.swift:1086:15: error: Objective-C method 'gestureRecognizer:shouldReceive:' provided by method 'gestureRecognizer(_:shouldReceive:)' does not match the requirement's selector ('gestureRecognizer:shouldReceiveTouch:')
A shell task (/usr/bin/xcrun xcodebuild -project /Carthage/Checkouts/Material/Material.xcodeproj -scheme "Material iOS" -configuration Release -sdk iphoneos ONLY_ACTIVE_ARCH=NO BITCODE_GENERATION_MODE=bitcode CODE_SIGNING_REQUIRED=NO CODE_SIGN_IDENTITY= CARTHAGE=YES clean build) failed with exit code 65:
** BUILD FAILED **


The following build commands failed:
	CompileSwift normal arm64
	CompileSwiftSources normal arm64 com.apple.xcode.tools.swift.compiler
(2 failures)
```

**Comment by ejimenez93** [comment]
Here's the output log:

```
GNU nano 2.0.6                                         File: carthage-xcodebuild.fKIVtj.log                                                                                        

write-file /Users/ejimenez/Library/Developer/Xcode/DerivedData/Material-hdqyzivytatgwncmeigeyqnhljnj/Build/Intermediates/Material.build/Release-iphoneos/Material\ iOS.build/Materia$
write-file /Users/ejimenez/Library/Developer/Xcode/DerivedData/Material-hdqyzivytatgwncmeigeyqnhljnj/Build/Intermediates/Material.build/Release-iphoneos/Material\ iOS.build/swift-o$
write-file /Users/ejimenez/Library/Developer/Xcode/DerivedData/Material-hdqyzivytatgwncmeigeyqnhljnj/Build/Intermediates/Material.build/Release-iphoneos/Material\ iOS.build/unexten$
write-file /Users/ejimenez/Library/Developer/Xcode/DerivedData/Material-hdqyzivytatgwncmeigeyqnhljnj/Build/Intermediates/Material.build/Release-iphoneos/Material\ iOS.build/unexten$
write-file /Users/ejimenez/Library/Developer/Xcode/DerivedData/Material-hdqyzivytatgwncmeigeyqnhljnj/Build/Intermediates/Material.build/Release-iphoneos/Material\ iOS.build/Materia$
write-file /Users/ejimenez/Library/Developer/Xcode/DerivedData/Material-hdqyzivytatgwncmeigeyqnhljnj/Build/Intermediates/Material.build/Release-iphoneos/Material\ iOS.build/Objects$

Create product structure
/bin/mkdir -p /Users/ejimenez/Library/Developer/Xcode/DerivedData/Material-hdqyzivytatgwncmeigeyqnhljnj/Build/Products/Release-iphoneos/Material.framework/Modules
/bin/mkdir -p /Users/ejimenez/Library/Developer/Xcode/DerivedData/Material-hdqyzivytatgwncmeigeyqnhljnj/Build/Products/Release-iphoneos/Material.framework/Headers

CompileSwiftSources normal arm64 com.apple.xcode.tools.swift.compiler
    cd /Carthage/Checkouts/Material
    export DEVELOPER_DIR=/Applications/Xcode.app/Contents/Developer
    export PATH="/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin:/Applications/Xcode.app/Contents/Developer/usr/bin:/Users/ejimenez/bin:/op$
    export SDKROOT=/Applications/Xcode.app/Contents/Developer/Platforms/iPhoneOS.platform/Developer/SDKs/iPhoneOS10.0.sdk
    export TOOLCHAINS=com.apple.dt.toolchain.XcodeDefault
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swiftc -incremental -module-name Material -O -whole-module-optimization -sdk /Application$

CompileSwift normal arm64
    cd /Carthage/Checkouts/Material
    /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/swift -frontend -emit-bc /Users/ejimenez/Documents/Retrographic/Sprocket/Carthage/Checkou$
/Users/ejimenez/Documents/Retrographic/Sprocket/Carthage/Checkouts/Material/Sources/iOS/NavigationDrawerController.swift:1086:15: error: Objective-C method 'gestureRecognizer:shoul$
    open func gestureRecognizer(_ gestureRecognizer: UIGestureRecognizer, shouldReceive touch: UITouch) -> Bool {
```

**Comment by ejimenez93** [comment]
Hey @danieldahan,

Think I might have found the issue. You might need to add this to your last commit (line 1085 of NavigationDrawerController.swift):

`@objc(gestureRecognizer:shouldReceiveTouch:)` instead of just @objc

Let me know if that helps!

**Comment by danieldahan** [comment]
Fixed and released in [Material 2.3.12](https://github.com/CosmicMind/Material/releases/tag/2.3.12).
